### DISCUSSION TITLE
In the Reference Architecture, can I run `terragrunt run-all <command>` from the root?

### DISCUSSION THREAD

**Discussion by rhoboat** [gruntwork-io/knowledge-base#96] [Help] [answered]
Should I be able to run `terragrunt plan` or `terragrunt run-all plan` from the root of the `infrastructure-live` directory?

**Comment by rhoboat** [comment, marked as answer]
No, and this is because Terragrunt cannot access multiple accounts at the same time and can only assume one role at a time. The best you can do is run `terragrunt run-all <command>` from the root of the account directory, i.e., security, shared, logs, dev, stage, prod.

Please also take a look at this [comprehensive answer](https://github.com/gruntwork-io/terragrunt/issues/720#issuecomment-497888756) on running terragrunt manually/locally, including the `run-all` commands.

**Comment by brikis98** [comment]
Also, bear in mind that `run-all plan` doesn't work very well if 

- (a) you have `dependency` blocks in your Terragrunt config and 
- (b) you hadn't previously run `apply` on all the modules. 

That's because Terragrunt will try to read the output variables for those `dependency` blocks. But if you hadn't run `apply`, there won't be any output variables, and it'll fail. Terraform can natively generate a plan when the outputs are not known by piping "known at apply time" placeholders through its dependency graph, but Terragrunt currently has no way of doing the same.

So, we only recommend using `run-all plan` on an infrastructure that has _already been deployed_, mostly as a way to see what changes there are that may need to be incrementally applied after.
